src/databricks/labs/dqx/profiler/profiler_runner.py

```diff
@@ -50,15 +50,27 @@ def run(
         Returns:
             A tuple containing the generated checks and profile summary statistics.
         """
-        df = read_input_data(self.spark, input_config)
-        summary_stats, profiles = self.profiler.profile(
-            df,
-            options={
-                "sample_fraction": profiler_config.sample_fraction,
-                "sample_seed": profiler_config.sample_seed,
-                "limit": profiler_config.limit,
-            },
-        )
+        # Build common profiling options
+        options = {
+            "sample_fraction": profiler_config.sample_fraction,
+            "sample_seed": profiler_config.sample_seed,
+            "limit": profiler_config.limit,
+            # LLM-based Primary Key Detection Options (available for both tables and DataFrames)
+            "enable_llm_pk_detection": profiler_config.llm_config.enable_pk_detection,
+            "llm_pk_detection_endpoint": profiler_config.llm_config.pk_detection_endpoint,
+            "llm_pk_validate_duplicates": True,  # Always validate duplicates for PK detection
+            "llm_pk_max_retries": 3,  # Fixed to 3 retries for optimal performance
+        }
+
+        # Use table-specific profiling for tables and file paths (enables enhanced LLM detection)
+        if hasattr(input_config, 'location') and input_config.location:
```

Comment:
pls remove this and leave how it was before with just calling: self.profiler.profile